### PR TITLE
Fixes #6768: Sometimes, promises get corrupted to the point they can't be automatically repaired on a node

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/6768

### CODE REVIEW

File: rudder-agent/SOURCES/check-rudder-agent [hunk 1]
```diff
@@ -241,3 +241,12 @@ if [ "z${OS_FAMILY}" != "zAIX" ]; then
   check_and_fix_specific_rudder_agent_file /etc/default/rudder-agent default
   check_and_fix_specific_rudder_agent_file /etc/cron.d/rudder-agent cron
 fi
+
+# if file is absent or empty there have been a problem with update
+base="/var/rudder/cfengine-community/inputs"
+if [ ! -f "${base}/common/1.0/update.cf" ] || [ $(stat -c "%s" "${base}/common/1.0/update.cf") = 0 ] || [ ! -f "${base}/failsafe.cf" ] || [ $(stat -c "%s" "${base}/failsafe.cf") = 0 ]
```
Comment (ncharles) [inline rudder-agent/SOURCES/check-rudder-agent]: you ought to check promises.cf; if it is not there, cf-execd may never run and restore the situation

File: rudder-agent/SOURCES/check-rudder-agent [hunk 2]
```diff
@@ -241,3 +241,12 @@ if [ "z${OS_FAMILY}" != "zAIX" ]; then
...
+then
+  rm -rf /var/rudder/cfengine-community/inputs/*
+  cp -r /opt/rudder/share/initial-promises/* /var/rudder/cfengine-community/inputs
+  /var/rudder/cfengine-community/bin/cf-agent -K -f failsafe.cf && /var/rudder/cfengine-community/bin/cf-agent
```
Comment (ncharles) [inline rudder-agent/SOURCES/check-rudder-agent]: you are missing some echo_n "ERROR: The promises files are missing. Restoring them from the initial promises."
echo " Done"

**Comment by ncharles** [comment]
once comments are adressed, you can self merge

**Comment by peckpeck** [comment]
PR updated

**Comment by ncharles** [comment]
@peckpeck you forgot the echo "Done" at the end. and it should have been echo_n for the first text

**Pull request merged**